lib/src/test/java/io/ably/lib/test/realtime/RealtimePresenceTest.java

```diff
@@ -1941,4 +1941,188 @@ public void realtime_presence_get_throws_when_channel_failed() throws AblyExcept
 				ably.close();
 		}
 	}
+
+	/**
+	 * Test if after reattach when returning from suspended mode client re-enters the channel with the same data
+	 * @throws AblyException
+	 *
+	 * Tests RTP17
+	 */
+	@Test
+	public void realtime_presence_suspended_reenter() throws AblyException {
+		AblyRealtime ably = null;
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = testVars.createOptions(testVars.keys[0].keyStr);
+			ably = new AblyRealtime(opts);
+
+			ConnectionWaiter connectionWaiter = new ConnectionWaiter(ably.connection);
+			connectionWaiter.waitFor(ConnectionState.connected);
+
+			final boolean[] presenceWaiter = new boolean[] {false};
+
+			Channel channel = ably.channels.get("presence_suspended_reenter");
+			channel.attach();
+			ChannelWaiter channelWaiter = new ChannelWaiter(channel);
+
+			final String presenceData = "PRESENCE_DATA";
+
+			channelWaiter.waitFor(ChannelState.attached);
+			channel.presence.enterClient(testClientId1, presenceData, new CompletionListener() {
+				@Override
+				public void onSuccess() {
+					synchronized (presenceWaiter) {
+						presenceWaiter[0] = true;
+						presenceWaiter.notify();
+					}
+				}
+
+				@Override
+				public void onError(ErrorInfo reason) {
+					fail("Presence: enter() failed");
+				}
+			});
+
+			try {
+				synchronized (presenceWaiter) {
+					while (!presenceWaiter[0])
+						presenceWaiter.wait();
+				}
+			}
+			catch (InterruptedException e) {
+			}
+
+			ably.connection.connectionManager.requestState(ConnectionState.suspended);
```

Comment:
I think this sequence only really affects the client state, and doesn't do anything to the server state. So when you later check that the server has the expected presence data, that doesn't confirm that the client re-entered; the server might just have the same presence members from when the client first entered.

Is it possible to modify the data cached in the internal presence map, so that updated member is reentered with the channel is re-attached, and you can confirm that it was updated on the server?